test/suite/hook_disagg.py

```diff
@@ -0,0 +1,390 @@
+#
+# Substitute oligarch tables for regular (row-store) tables in Python tests.
+#
+# These hooks are intended for basic testing.  There are several
+# pieces of functionality here.
+#
+# 1. We add disaggregated storage parameters to the config when calling wiredtiger_open()
+#
+# 2. When creating a table, we add options to use the page_log service.
+#
+# 3. We stub out some functions that aren't supported by disaggregated storage.
+#
+# To run, for example, the cursor tests with these hooks enabled:
+#     ../test/suite/run.py --hook disagg cursor
+#
+from __future__ import print_function
+
+import os, re, unittest, wthooks, wttest
+from wttest import WiredTigerTestCase
+from helper_disagg import DisaggConfigMixin, gen_disagg_storages
+
+# These are the hook functions that are run when particular APIs are called.
+
+# Add the local storage extension whenever we call wiredtiger_open
+def wiredtiger_open_disagg(ignored_self, args):
+
+    disagg_storage_sources = gen_disagg_storages()
+    testcase = WiredTigerTestCase.currentTestCase()
+    extension_name = testcase.getDisaggService()
+    extension_config = testcase.getDisaggConfig()
+
+    valid_storage_source = False
+    # Construct the configuration string based on the storage source.
+    for disagg_details in disagg_storage_sources:
+        if (testcase.getDisaggService() == disagg_details[0]):
+            valid_storage_source = True
+
+    if valid_storage_source == False:
+        raise AssertionError('Invalid storage source passed in the argument.')
+
+    curconfig = args[-1]
+    homedir = args[0]
+
+    # If there is already disagg storage enabled, we shouldn't enable it here.
+    # We might attempt to let the wiredtiger_open complete without alteration,
+    # however, we alter several other API methods that would do weird things with
+    # a different disagg_storage configuration. So better to skip the test entirely.
+    if 'oligarch=' in curconfig:
+        skip_test("cannot run disagg hook on a test that already uses disagg storage")
+
+    # Similarly if this test is already set up to run disagg vs non-disagg scenario, let's
+    # not get in the way.
+    if hasattr(testcase, 'disagg_conn_config'):
+        skip_test("cannot run disagg hook on a test that already includes DisaggConfigMixin")
+
+    if 'in_memory=true' in curconfig:
+        skip_test("cannot run disagg hook on a test that is in-memory")
+
+    # Mark this test as readonly, but don't disallow it.  See testcase_is_readonly().
+    if 'readonly=true' in curconfig:
+        testcase._readonlyDisaggTest = True
+
+    extension_libs = WiredTigerTestCase.findExtension('page_log', extension_name)
+    if len(extension_libs) == 0:
+        raise Exception(extension_name + ' storage source extension not found')
+
+    disagg_config = ',log=(enabled),transaction_sync=(enabled,method=fsync),statistics=(all),statistics_log=(wait=1,json=true,on_close=true)'
+    disagg_config += ',oligarch=(role="leader")'
```

Comment:
Also need to set up `disaggregated=(stable_prefix=.,page_log=palm)` here. This is now required at the connection level.